plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/SpreadsheetPresentation.java

```diff
@@ -848,7 +849,9 @@ public void refreshData(boolean refreshMetadata, boolean append, boolean keepSta
                 controller.getDataContainer().isFeatureSupported(DBSDataContainer.FEATURE_DATA_FILTER) &&
                 controller.getPreferenceStore().getBoolean(ResultSetPreferences.RESULT_SET_SHOW_ATTR_FILTERS);
         autoFetchSegments = controller.getPreferenceStore().getBoolean(ResultSetPreferences.RESULT_SET_AUTO_FETCH_NEXT_SEGMENT);
-        calcColumnWidthByValue = getPreferenceStore().getBoolean(ResultSetPreferences.RESULT_SET_CALC_COLUMN_WIDTH_BY_VALUES);
+        calcColumnWidthMethod = CommonUtils.valueOf(ResultSetPreferences.GridColumnCalcWidthMethod.class,
+                getPreferenceStore().getString(ResultSetPreferences.RESULT_SET_CALC_COLUMN_WIDTH_METHOD));
```

Comment:
Please pass the default value to the `CommonUtils.valueOf`, otherwise it will be `null` if it can't be parsed.